enterprise/views.py

```diff
@@ -1462,7 +1464,7 @@ def get(self, request, *args, **kwargs):
 
         # Directly enroll in audit mode if the request in question has full direct audit enrollment eligibility.
         resource_id = course_run_id or program_uuid
-        if self.eligible_for_direct_audit_enrollment(request, enterprise_customer, resource_id):
+        if self.eligible_for_direct_audit_enrollment(request, enterprise_customer, resource_id, course_key):
```

Comment:
Can we just update above line with 
```python
resource_id = course_key or course_run_id or program_uuid
```
instead of updating the method `eligible_for_direct_audit_enrollment`?

Reply:
No, Course_run_id is in use in that method. this may not work for the actual enroll function or similar right below it though.

Reply:
This code is very confusing given that we're passing `resource_id` which may be a `program_uuid`, but in `eligible_for_direct_audit_enrollment` the parameter is just called `course_run_id`

I know you didn't update this but I'm actually surprised the other 2 method calls inside `eligible_for_direct_audit_enrollment ` that use this work with programs: (`enterprise_customer.catalog_contains_course` and `EnrollmentApiClient().has_course_mode`)

If we are passing around course_run_id or program_uuid I'd recommend updating the variable name in `eligible_for_direct_audit_enrollment` to `resource_id` with a comment in the method that `resource_id` is `course_run_id` or `program_uuid` to avoid this confusion in the future.  It doesn't look like this gets used anywhere other than this call.